primitives/arithmetic/src/per_things.rs

```diff
@@ -1883,6 +1883,28 @@ macro_rules! implement_per_thing_with_perthousand {
 	}
 }
 
+impl Into<Perbill> for I64F64 {
+	fn into(self) -> Perbill {
+		if self <= I64F64::from_num(0i32) {
+			Perbill::zero()
+		} else if self >= I64F64::from_num(1i32) {
+			Perbill::one()
+		} else {
+			let per_bill = self
+				.checked_mul(I64F64::from_num(1_000_000_000i32))
+				.expect("0 < self < 1, cannot overflow; qed");
+			let rounded: i64 = per_bill.round().lossy_into();
+			Perbill::from_parts(rounded as u32)
+		}
+	}
+}
+
+impl From<Perbill> for I64F64 {
+	fn from(a: Perbill) -> Self {
+		I64F64::from_num(a.deconstruct() as u128).saturating_div(I64F64::from_num(1_000_000_000i32))
```

Comment:
What type  is perbill? I just want to make sure that `as u128` can't panic.

Reply:
it represents a int value in [0, 10^9] which will easily fit into a u128.
i took the pattern from [here](https://github.com/pifragile/substrate/blob/48b878cce920a6642c256830b065143e9b299d20/primitives/arithmetic/src/fixed_point.rs#L447)

Reply:
Ok, perfect. Thanks for explaining!